tools/boost-builder/Dockerfile

```diff
@@ -0,0 +1,9 @@
+# Does this match Travis's trusty?
+FROM ubuntu:trusty
+
+ADD build_boosts.zsh /root/
+
+RUN apt-get update
+RUN apt-get -y install build-essential python2.7-dev wget zsh
```

Comment:
Checked a few random system libs and they're compiled with the exact gcc that build-essential ships. I'm satisfied; happy to keep digging if you aren't.